PR: https://github.com/odoo/odoo/pull/126438

From: 45ac2970b00b352b86b1fbe6f8b25c15909c6d04
From: dasz
Date: 2023-09-21 07:40:04

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = 'hr.resume.line'
-    display_type = fields.Selection(selection_add=[('certification', 'Certification')])
-    survey_id = fields.Many2one('survey.survey', string='Certification', readonly=True)

Total Changes: 35

[IMP] hr_skills_survey: add certifications report

This report allows an HR manager to quickly see the status of employee's certifications
red: expired, yellow: expiring in 3 months or less. There are also 2 filters for those
situations.

task 3384857

closes odoo/odoo#126438

Signed-off-by: Sofie Gvaladze (sgv) <sgv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_skills_survey/models/__init__.py
+++ b/addons/hr_skills_survey/models/__init__.py
@@ -2,3 +2,4 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import survey_user
+from . import hr_resume_line

--- a/None
+++ b/addons/hr_skills_survey/models/hr_resume_line.py
@@ -0,0 +1,27 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class ResumeLine(models.Model):
+    _inherit = 'hr.resume.line'
+
+    display_type = fields.Selection(selection_add=[('certification', 'Certification')])
+    department_id = fields.Many2one(related="employee_id.department_id", store=True)
+    survey_id = fields.Many2one('survey.survey', string='Certification', readonly=True)
+    expiration_status = fields.Selection([
+        ('expired', 'Expired'),
+        ('expiring', 'Expiring'),
+        ('valid', 'Valid')], compute='_compute_expiration_status', store=True)
+
+    @api.depends('date_end')
+    def _compute_expiration_status(self):
+        self.expiration_status = 'valid'
+        for line in self:
+            if line.date_end:
+                if line.date_end <= fields.Date.today():
+                    line.expiration_status = 'expired'
+                elif line.date_end + relativedelta(months=-3) <= fields.Date.today():
+                    line.expiration_status = 'expiring'

--- a/addons/hr_skills_survey/models/survey_user.py
+++ b/addons/hr_skills_survey/models/survey_user.py
@@ -32,10 +32,3 @@ class SurveyUserInput(models.Model):
                 'display_type': 'certification',
                 'survey_id': survey.id
             })
-
-
-class ResumeLine(models.Model):
-    _inherit = 'hr.resume.line'
-
-    display_type = fields.Selection(selection_add=[('certification', 'Certification')])
-    survey_id = fields.Many2one('survey.survey', string='Certification', readonly=True)
